python_modules/dagster/dagster/core/execution_plan/multiprocessing_engine.py

```diff
@@ -111,42 +145,50 @@ def __init__(self, pipeline_fn):
         self.pipeline_fn = pipeline_fn
 
 
+def _all_inputs_covered(step, intermediates_manager):
+    for step_input in step.step_inputs:
+        if not intermediates_manager.has_value(step_input.prev_output_handle):
+            return False
+    return True
+
+
 def multiprocess_execute_plan(executor_config, pipeline_context, execution_plan):
     check.inst_param(executor_config, 'executor_config', MultiprocessExecutorConfig)
     check.inst_param(pipeline_context, 'pipeline_context', PipelineExecutionContext)
     check.inst_param(execution_plan, 'execution_plan', ExecutionPlan)
 
     step_levels = execution_plan.topological_step_levels()
 
+    # TODO where to put in xplat way?
+    # TODO probably manage in pipeline execution context?
+    root_dir = '/tmp/dagster/runs/{run_id}'.format(run_id=pipeline_context.run_id)
```

Comment:
this won't work on windows -- we might want to let people config the actual root (`/`) with an env variable & sensible defaults for each platform (`C:\tmp\...` etc) or use `tempfile.mkdtemp`

Reply:
will address when we do files resource

Reply:
fwiw, this is a claim about how python looks for tempdirs to use with `mkdtemp`: https://stackoverflow.com/a/847866/324449